### ISSUE TITLE
[0.43.4] Artillery base TN being restricted to either 7 or 8 when generating units for MM

### ISSUE THREAD

**Issue by RexPearce** [MegaMek/mekhq#507]
I noticed my artillery units were using a base TN of 7, despite having a crew member with an artillery base TN of 4. At first I though this was crew averaging, but even giving all crew member artillery skills still results in a TN of 7 in MM (Or a TN of 8 if no crew members haver artillery skill).

I dug around the code, and I think the cause is line 2895 of Unit.java:
```javascript
        entity.getCrew().setPiloting(Math.min(Math.max(piloting, 0), 8), 0);
        entity.getCrew().setGunnery(Math.min(Math.max(gunnery, 0), 7), 0);
        entity.getCrew().setArtillery(Math.min(Math.max(artillery, 7), 8), 0); #  < TYPO?
```

Should the last line above be: entity.getCrew().setArtillery(Math.min(Math.max(artillery, **0**), 8), 0); ?

**Comment by mkerensky** [comment]
This sounds normal. Artillery is supposed to have a base TN of 7 for indirect fire, + gunnery/artillery skill and other modifiers from there.  If what you're trying to say is that the unit was given an artillery skill of 7 and was needing TNs of 14 to hit a hex, that's a problem.

**Comment by RexPearce** [comment]
> If what you're trying to say is that the unit was given an artillery skill of 7 and was needing TNs of 14 to hit a hex, that's a problem.

That's exactly the behavior I'm describing.

With an entire crew having elite (base +2) artillery skills, I'm getting a base skill TN of 7, and an additional + 7 for indirect fire, making the hit impossible.

**Comment by Qwertronix** [comment]
Will take a shot at fixing this tonight since it annoys me as well.

**Comment by NickAragua** [comment]
Fixed with PR #522
